metals/src/main/scala/scala/meta/internal/troubleshoot/ScalaProblem.scala

```diff
@@ -0,0 +1,75 @@
+package scala.meta.internal.troubleshoot
+
+import scala.meta.internal.metals.BuildInfo
+import scala.meta.internal.metals.ScalaVersions
+
+sealed abstract class ScalaProblem {
+  def message: String
+
+  protected val hint = "run 'Build import' to enable code navigation."
+
+}
+
+// scala version issues
+case class UnsupportedScalaVersion(version: String) extends ScalaProblem {
+  override def message: String = {
+    if (ScalaVersions.isSupportedScalaBinaryVersion(version)) {
+      val recommended = ScalaVersions.recommendedVersion(version)
+      s"Upgrade to Scala $recommended and " + hint
+    } else {
+      val versionToUpgradeTo =
+        if (ScalaVersions.isScala3Version(version)) {
+          s"Scala ${BuildInfo.scala3}"
+        } else {
+          s"Scala ${BuildInfo.scala213} or ${BuildInfo.scala212}"
+        }
+      s"Code navigation is not supported for this compiler version, change to " + versionToUpgradeTo + " and " + hint
+    }
+  }
+}
+
+case class DeprecatedScalaVersion(version: String) extends ScalaProblem {
+  override def message: String =
+    s"Scala $version might not be supported in upcoming versions of Metals, " +
+      s"please upgrade to Scala ${ScalaVersions.recommendedVersion(version)}."
+}
+
+case class FutureScalaVersion(version: String) extends ScalaProblem {
+  override def message: String = s"Scala $version is not yet supported"
+}
+
+// semanticdb problems
+case class SemanticDBDisabled(
+    scalaVersion: String,
+    bloopVersion: String,
+    isBloopVersionSupported: Boolean
+) extends ScalaProblem {
+  override def message: String = {
+    if (isBloopVersionSupported) {
+      s"""|The installed Bloop server version is $bloopVersion while Metals requires at least Bloop version ${BuildInfo.bloopVersion},
+          |To fix this problem please update your Bloop server.""".stripMargin
+    } else if (ScalaVersions.isSupportedScalaVersion(scalaVersion)) {
+      hint.capitalize
+    } else {
+      "Enable semanticdb plugin for the project."
```

Comment:
I'm not 100% sure on the situation on when this will be hit, but I think the more info we can give here the better, since for a lot of people this may not be clear. They might want to why this is being shown. Maybe this will just be combined with other messages, and if so ignore this, but I think having a bit more info, something like "Semanticdb is required for code navigation to work correctly in your project, however the semanticdb plugin doesn't seem to be enabled for your project. Please enable the semanticdb plugin for this project in order for code navigation to work correctly." wdyt?

Reply:
That should almost never happen, but your message is a much better one for sure.